server/pbench/bin/pbench-server-set-result-state

```diff
@@ -0,0 +1,41 @@
+#! /bin/bash
+
+###########################################################################
+# load common things
+opts=$SHELLOPTS
+case $opts in
+    *xtrace*)
+        dir=$(dirname $(which $0))
+        PROG=$(basename $(which $0))
+        ;;
+    *)
+        dir=$(dirname $0)
+        PROG=$(basename $0)
+        ;;
+esac
+
+. $dir/pbench-base.sh
+###########################################################################
+
+results_full_path=$1
+tarball=$2
+
+# symlink into the TODO state dir
+mkdir -p $results_full_path/TODO
+cd $results_full_path/TODO
+ln -s $results_full_path/$tarball .
+
+# we will get rid of this after a full release cycle,
+# thereby allowing the following code to run. See
+# issue #786.
+exit 0
+
+# add entry to watched file for inotify service
+mkdir -p ${INOTIFY_STATE_DIR}
+watched_file=${INOTIFY_STATE_DIR}/TO-DISPATCH-LIST
```

Comment:
Overall everything looks good to me but if we gonna use the self.name parameter i.e basically the name of the script in pbench-inotify script then I think this watched_file=${INOTIFY_STATE_DIR}/TO-DISPATCH-LIST path should be watched_file=${INOTIFY_STATE_DIR}/pbench-dispatch/TO-DISPATCH-LIST

If we are not using self.name then the code LGTM

Reply:
You are right: we won't use the script name in the state directory.